src/Microsoft.AspNet.Http/Features/QueryFeature.cs

```diff
@@ -41,13 +42,22 @@ public IReadableStringCollection Query
                 }
 
                 var queryString = _request.Fetch(_features).QueryString;
-                if (_query == null || _queryString != queryString)
+                if (!string.Equals(_queryString, queryString, StringComparison.Ordinal) || _query == null)
```

Comment:
Consider putting null check before string comparison. If you make that change, in case of _query being null, there wouldn't be a string comparison at all.